netket/nn/symmetric_linear.py

```diff
@@ -105,14 +65,18 @@ class DenseSymmMatrix(Module):
     precision: Any = None
     """numerical precision of the computation see `jax.lax.Precision`for details."""
 
-    kernel_init: NNInitFunc = default_densesymm_initializer
-    """Initializer for the Dense layer matrix. Defaults to variance scaling"""
+    kernel_init: NNInitFunc = default_gcnn_initializer
+    """Initializer for the kernel. Defaults to Lecun normal."""
     bias_init: NNInitFunc = zeros
-    """Initializer for the bias. Defaults to zero initialization"""
+    """Initializer for the bias. Defaults to zero initialization."""
 
     def setup(self):
         # pylint: disable=attribute-defined-outside-init
         self.n_symm, self.n_sites = np.asarray(self.symmetries).shape
+        if self.mask is not None:
+            self.scaled_mask = (
+                self.mask / jnp.linalg.norm(self.mask) * self.n_sites ** 0.5
+            )
```

Comment:
The mask must be an hashablearray, right? `jnp.linalg.norm(self.mask)` works because it's doing essentially `jnp.linalg.norm(jnp.array(np.asarray(self.mask)))`. 
But you are doing

```python
self.mask / jnp.linalg.norm(Self.mask)
```
which is converted to
```
jnp.array(self.mask) / jnp.linalg.norm(jnp.array(np.asarray(self.mask)))
```
which captures the mask twice.

I think a better implementation would be to capture only the rescaled mask once:

```suggestion
        if self.mask is not None:
            self.scaled_mask = jnp.asarray(
                self.mask / np.linalg.norm(self.mask) * self.n_sites ** 0.5)
```
(probably you need to import numpy and reformat too)

Reply:
Also can we assert that the bask is boolean and not integers or other things? 
Or do we need in some cases the mask to be a finite number?

Reply:
I've never used this feature so no strong feelings, but it might be useful to control the initialisation through this mask. The scaling logic is happy with that too.

Reply:
I generally don't like restricting inputs unless they break stuff/lead to something very absurd. This is neither.